### DISCUSSION TITLE
Build error with TS and Nextjs

### DISCUSSION THREAD

**Discussion by madeinspace** [react-hook-form/react-hook-form#8171] [General]
I'm getting that error trying to build my Nextjs app and wondering if you guys have a clue on how to fix it:

Failed to compile.

C:/Users/username/Documents/Workspace/myapp/node_modules/react-hook-form/dist/types/path/common.d.ts
ERROR in C:/Users/username/Documents/Workspace/myapp/node_modules/react-hook-form/dist/types/path/common.d.ts(39,53):
39:53 '?' expected.
    37 |  * @typeParam T - type which may be converted to a {@link Key}
    38 |  */
  > 39 | export declare type ToKey<T> = T extends ArrayKey ? `${T}` : AsKey<T>;
       |                                                     ^

"react-hook-form": "^7.29.0",
"typescript": "^4.3",

Can't deploy my app until I find a fix! thanks

**Comment by MagnusSafty** [comment]
Did you find the problem? I am facing the same issue

**Comment by MagnusSafty** [comment]
The issue was caused by an outdated typescript compiler, updated it and it all worked.
